mq/plugins/ldap_fixup.py

```diff
@@ -0,0 +1,31 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+from mozdef_util.utilities.key_exists import key_exists
+
+
+class message(object):
+    def __init__(self):
+        '''
+        rewrites ldap's details.tls field and sets source
+        '''
+
+        self.registration = ['ldapChange', 'ldif', 'LDAP-Humanizer']
+        self.priority = 5
+
+    def onMessage(self, message, metadata):
+
+        # check for category like 'ldap'
+        if key_exists('category', message):
+            if message.get['category'] == "ldap":
+                if key_exists('details.tls', message):
+                    message['details.tls_encrypted'] = message['details.tls']
+                    del message['details']['tls']
+
+        if 'source' not in message:
+            message['source'] = "ldap"
+            return (message, metadata)
```

Comment:
This return statement isn't necessary since we'll hit the one directly under this.

Reply:
removed! ty!